cfme/tests/openstack/cloud/test_instances.py

```diff
@@ -0,0 +1,151 @@
+
+
+def test_pause_instance(new_instance):
+    new_instance.power_control_from_cfme(from_details=True,
+                                         option=OpenStackInstance.PAUSE)
+    new_instance.wait_for_instance_state_change(OpenStackInstance.STATE_PAUSED)
+    state = new_instance.get_detail(properties=('Power Management',
+                                                'Power State'))
+    assert state == OpenStackInstance.STATE_PAUSED
+
+
+def test_shelve_instance(new_instance):
+    new_instance.power_control_from_cfme(from_details=True,
+                                         option=OpenStackInstance.SHELVE)
+    new_instance.wait_for_instance_state_change(OpenStackInstance.STATE_SHELVED)
+    state = new_instance.get_detail(properties=('Power Management',
+                                                'Power State'))
+    assert state in (OpenStackInstance.STATE_SHELVED_OFFLOAD,
+                     OpenStackInstance.STATE_SHELVED)
+
+
+def test_shelve_offload_instance(shelved_instance):
+    shelved_instance.power_control_from_cfme(from_details=True,
+                                             option=OpenStackInstance.SHELVE_OFFLOAD)
+    shelved_instance.wait_for_instance_state_change(
+        OpenStackInstance.STATE_SHELVED_OFFLOAD)
+    state = shelved_instance.get_detail(properties=('Power Management',
+                                                    'Power State'))
+    assert state == OpenStackInstance.STATE_SHELVED_OFFLOAD
+
+
+def test_start_instance(shelved_instance):
+    shelved_instance.power_control_from_cfme(from_details=True,
+                                             option=OpenStackInstance.START)
+    shelved_instance.wait_for_instance_state_change(OpenStackInstance.STATE_ON)
+    state = shelved_instance.get_detail(properties=('Power Management',
+                                                    'Power State'))
+    assert state == OpenStackInstance.STATE_ON
+
+
+def test_soft_reboot_instance(new_instance):
+    new_instance.power_control_from_cfme(from_details=True,
+                                         option=OpenStackInstance.SOFT_REBOOT)
+    new_instance.wait_for_instance_state_change(OpenStackInstance.STATE_REBOOTING)
+
+    state = new_instance.get_detail(properties=('Power Management', 'Power State'))
+    assert state in (OpenStackInstance.STATE_ON,
+                     OpenStackInstance.STATE_REBOOTING)
+
+
+def test_hard_reboot_instance(new_instance):
+    new_instance.power_control_from_cfme(from_details=True,
+                                         option=OpenStackInstance.HARD_REBOOT)
+    new_instance.wait_for_instance_state_change(OpenStackInstance.STATE_REBOOTING)
+
+    state = new_instance.get_detail(properties=('Power Management',
+                                                'Power State'))
+    assert state in (OpenStackInstance.STATE_ON,
+                     OpenStackInstance.STATE_REBOOTING)
+
+
+def test_delete_instance(new_instance):
```

Comment:
`TERMINATE ` is just a name of var, which can point on "Delete" or "Terminate" power option on the page, it depends on CF version. I've named test as delete because this is the option which displayed in latest versions.